test-sbt/shared/src/test/scala/zio/test/sbt/ExecuteSpecs.scala

```diff
@@ -0,0 +1,56 @@
+package zio.test.sbt
+
+import sbt.testing.{Event, EventHandler, Selector, SuiteSelector, TaskDef}
+import zio.ZIO
+import zio.test.{ZIOSpecAbstract, testConsole}
+
+import scala.collection.mutable.ArrayBuffer
+
+object ExecuteSpecs {
+  def getOutput(
+    spec: ZIOSpecAbstract,
+    args: Array[String] = Array.empty
+  ): ZIO[Any, Throwable, Seq[String]] =
+    getOutputs(Seq(spec), args).mapError(_.head)
+
+  def getOutputs(
+    specs: Seq[ZIOSpecAbstract],
+    args: Array[String] = Array.empty
+  ): ZIO[Any, ::[Throwable], Seq[String]] =
+    getOutputsAndEvents(specs, args).map(_._1)
+
+  def getEvents(
+    spec: ZIOSpecAbstract,
+    args: Array[String] = Array.empty,
+    selectors: Array[Selector] = Array(new SuiteSelector)
+  ): ZIO[Any, ::[Throwable], Seq[Event]] =
+    getOutputsAndEvents(Seq(spec), args, selectors).map(_._2)
+
+  def getOutputsAndEvents(
+    specs: Seq[ZIOSpecAbstract],
+    args: Array[String],
+    selectors: Array[Selector] = Array(new SuiteSelector)
+  ): ZIO[Any, ::[Throwable], (Seq[String], Seq[Event])] = {
```

Comment:
Since we're initializing a mutable object inside, we need to be referentially transparent
 
```suggestion
  ): ZIO[Any, ::[Throwable], (Seq[String], Seq[Event])] =  ZIO.suspendSucceed {
```

Reply:
Thanks! Fixing it.

Reply:
done in https://github.com/zio/zio/pull/10120/commits/90bfe810a6d3ec4f2d16108ca4329dfb06fb77e7